apps/wing-language-server/src/prep.rs

```diff
@@ -31,7 +31,7 @@ pub fn parse_text(source_file: &str, text: &[u8]) -> ParseResult {
 	let mut scope = wing_parser.wingit(&tree.root_node());
 
 	let mut types = type_check::Types::new();
-	let type_diag = type_check(&mut scope, &mut types);
+	let type_diag = type_check(&mut scope, &mut types, &Path::new(source_file));
```

Comment:
This path will not be absolute, so this will fail with the newly added is_absolute checks. 
Technically this doesn't even have to be a file path, it can be an arbitrary uri.

Reply:
Got it. So, we need an absolute source file path is so that the compiler can locate JSII libraries in the file system and load their .jsii files with wingii etc. I'm sort of bundling two use cases here (one is to use the file name for error messages and diagnostics, the other is to look at the parent directories), but in either case I think it might simplify our design if we say that in our wingc-as-a-library, `compile` must be invoked with an absolute file path.

@MarkMcCulloh Is there a way we can convert the URI to a file path or just decide to throw if it can't be converted into one? (after all, I don't think we should accept arbitrary URIs like `tel:+1-555-555-5555` as input to our compiler)

Reply:
https://github.com/winglang/wing/blob/main/apps/wing-language-server/src/main.rs#L202

params.uri has a `to_file_path()` that should be good, so you could change that line to something like:

```rs
let parse_result = parse_text(
	params
		.uri
		.to_file_path()
		.expect("Only 'file' scheme supported")
		.to_str()
		.unwrap(),
	&params.text.as_bytes(),
);
```